optimum/intel/openvino/modeling_timm.py

```diff
@@ -0,0 +1,398 @@
+
+        self.config = config
+        if feature_only:
+            self.timm_model = timm.create_model(
+                "hf-hub:" + self.config.hf_hub_id, num_classes=0, pretrained=pretrained, in_chans=in_chans
+            )
+        else:
+            self.timm_model = timm.create_model(
+                "hf-hub:" + self.config.hf_hub_id,
+                num_classes=self.config.num_labels,
+                pretrained=pretrained,
+                in_chans=in_chans,
+            )
+        self.timm_model.eval()
+
+    @classmethod
+    def from_pretrained(cls, model_name_or_path, **kwargs):
+        config = TimmConfig.from_pretrained(model_name_or_path, **kwargs)
+        return cls(config, **kwargs)
+
+    def forward(
+        self,
+        pixel_values: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutput]:
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+
+        # TODO: maybe have a cleaner way to cast the input (from `ImageProcessor` side?)
+        # expected_dtype = self.embeddings.patch_embeddings.projection.weight.dtype
+        # if pixel_values.dtype != expected_dtype:
+        #     pixel_values = pixel_values.to(expected_dtype)
+
+        model_output = self.timm_model(pixel_values)
+
+        if not return_dict:
+            return model_output
+
+        return BaseModelOutput(last_hidden_state=model_output, hidden_states=None)
+
+
+class TimmForImageClassification(TimmPreTrainedModel):
+    def __init__(self, config: TimmConfig, num_labels: int = None, **kwargs) -> None:
+        super().__init__(config, **kwargs)
+
+        if num_labels:
+            config.num_labels = num_labels
+        self.timm = TimmModel(config, feature_only=False)
+
+    @classmethod
+    def from_pretrained(cls, model_name_or_path, **kwargs):
+        config = TimmConfig.from_pretrained(model_name_or_path, **kwargs)
+        return cls(config, **kwargs)
+
+    def forward(
+        self,
+        pixel_values: Optional[torch.Tensor] = None,
+        labels: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[tuple, ImageClassifierOutput]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size,)`, *optional*):
+            Labels for computing the image classification/regression loss. Indices should be in `[0, ...,
+            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
+            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        logits = self.timm(
+            pixel_values,
+            return_dict=return_dict,
+        )
+
+        loss = None
```

Comment:
Loss is not needed at all in the context of further export to OpenVINO.

Reply:
I have removed that section.